source/NVDAObjects/behaviors.py

```diff
@@ -390,8 +390,8 @@ def _get_caretMovementDetectionUsesEvents(self):
 		return False
 
 
-class KeyboardHandlerBasedTypedCharSupport(Terminal):
-	"""A Terminal object that also provides typed character support for
+class EnhancedTermTypedCharSupport(Terminal):

```

Comment:
Is it really necessary to rename this? It breaks compatibility.

Reply:
Compatibility with what?

Reply:
Backwards compatibility with older versions of NVDA. It is unlikely but possible that add-ons use this class.

Reply:
> Backwards compatibility with older versions of NVDA. It is unlikely but possible that add-ons use this class.

If having a class named `KeyboardHandlerBasedTypedCharSupport` that doesn't actually use `keyboardHandler` in some cases isn't that confusing, I can revert the rename.

Reply:
May be you could re-add KeyboardHandlerBasedTypedCharSupport as a subclass of EnhancedTermTypedCharSupport, it can just do nothing. This way, you could also remove the _shouldUseToUnicodeEx attribute, as the isinstance check should be sufficient.